internal/core/unittest/test_kmeans_clustering.cpp

```diff
@@ -34,6 +33,7 @@ ReadPBFile(std::string& file_path, google::protobuf::Message& message) {
     std::ifstream infile;
     infile.open(file_path.data(), std::ios_base::binary);
     if (infile.fail()) {
+        infile.close();
```

Comment:
Actually, there is no need to explicitly invoke the "close" function.
This is what RAII is for, let the destructor do its job. There is no harm in closing it manually, but it's not the C++ way.